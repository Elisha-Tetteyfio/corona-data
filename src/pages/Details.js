import React from 'react';
import { NavLink } from 'react-router-dom';
import styles from './Details.module.css';
import settings from '../images/settings.png';

const Details = (country) => {
  const { name, latest_data: latestData, population } = country;
  return (
    <>
      <header>
        <NavLink to="/" exact className={styles.back}>&lt; Home</NavLink>
        {name}
        <img src={settings} alt="settings" />
      </header>
      <div className={styles.intro}>
        {name}
        {' '}
        corona details
      </div>
      <ul className={styles.details}>
        <li className={styles.detailsElement}>
          <div>Country</div>
          <div>{name}</div>
        </li>
        <li className={styles.detailsElement}>
          <div>Population</div>
          <div>{population}</div>
        </li>
        <li className={styles.detailsElement}>
          <div>Confirmed cases</div>
          <div>{latestData.confirmed}</div>
        </li>
        <li className={styles.detailsElement}>
          <div>Critical cases</div>
          <div>{latestData.critical}</div>
        </li>
        <li className={styles.detailsElement}>
          <div>Confirmed deaths</div>
          <div>{latestData.deaths}</div>
        </li>
        <li className={styles.detailsElement}>
          <div>Recovered cases</div>
          <div>{latestData.recovered}</div>
        </li>
      </ul>
    </>
  );
};

export default Details;
