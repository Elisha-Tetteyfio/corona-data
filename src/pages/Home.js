import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import { NavLink } from 'react-router-dom';
import { v4 as uuidv4 } from 'uuid';
import CountryCard from '../components/CountryCard';
import styles from './Home.module.css';
import settings from '../images/settings.png';

const Home = () => {
  const countries = useSelector((data) => data.data);
  const [search, setSearch] = useState([]);
  const Search = search.toString().charAt(0).toLocaleUpperCase() + search.slice(1);
  const searchedCountries = countries.filter((country) => country.name.startsWith(Search));
  let totalCases = 0;
  let totalCountries = 0;
  let totalDeaths = 0;
  countries.map((c) => {
    totalCases += c.latest_data.confirmed;
    totalCountries += 1;
    totalDeaths += c.latest_data.deaths;
    return totalCases;
  });
  return (
    <div>
      <header className={styles.header}>
        Home
        <div>
          <input
            type="text"
            placeholder="Search country"
            value={search}
            onChange={(e) => setSearch(e.target.value)}
          />
          <img src={settings} alt="settings" />
        </div>
      </header>
      <div className={styles.intro}>
        <h1>Corona Virus Data</h1>
        <div className={styles.allData}>
          <div>
            <div className={styles.value}>{totalCountries}</div>
            <div className={styles.parameter}>Number of countries</div>
          </div>
          <div>
            <div className={styles.value}>{totalCases.toLocaleString()}</div>
            <div className={styles.parameter}>Total number of cases</div>
          </div>
          <div>
            <div className={styles.value}>{totalDeaths.toLocaleString()}</div>
            <div className={styles.parameter}>Total number of deaths</div>
          </div>
        </div>
      </div>
      <div className={styles.divide}>Stats by countries</div>
      <div className={styles.countriesContainer}>
        {searchedCountries.map((country) => (
          <NavLink to={`/details/${country.name}`} exact key={uuidv4()} className={styles.link}>
            <CountryCard
              name={country.name}
              latest_data={country.latest_data}
              key={uuidv4()}
            />
          </NavLink>
        ))}
      </div>
    </div>
  );
};

export default Home;
