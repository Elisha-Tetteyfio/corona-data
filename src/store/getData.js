import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';

const url = 'https://corona-api.com/countries';
const GET_DATA = 'coronoa-data/GET_DATA';

export const fetchData = createAsyncThunk(GET_DATA, async () => {
  const result = await fetch(url);
  const resultJson = await result.json();
  return resultJson.data;
});

const dataSlice = createSlice({
  name: 'corona-data',
  initialState: [{ name: '', code: '', latest_data: { confirmed: '', deaths: '', recovered: '' } }],
  extraReducers: {
    [fetchData.fulfilled]: (state, action) => action.payload,
  },
});

// export const { } = dataSlice.actions;
export default dataSlice.reducer;
