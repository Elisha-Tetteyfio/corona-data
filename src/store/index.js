import { configureStore } from '@reduxjs/toolkit';
import data from './getData';

const store = configureStore({
  reducer: {
    data,
  },
});

export default store;
