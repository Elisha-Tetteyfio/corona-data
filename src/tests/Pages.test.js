import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import Details from '../pages/Details';
import store from '../store/index';
import Home from '../pages/Home';

describe('Testing pages', () => {
  it('test Details page', () => {
    const tree = renderer
      .create(
        <Router>
          <Details />
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('test Home page', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Home />
          </Router>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
