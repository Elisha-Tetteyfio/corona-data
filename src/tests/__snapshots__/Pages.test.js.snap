// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing pages test Details page 1`] = `
Array [
  <header>
    <a
      aria-current="page"
      className="back active"
      href="/"
      onClick={[Function]}
    >
      &lt; Home
    </a>
    <img
      alt="settings"
      src="settings.png"
    />
  </header>,
  <div
    className="intro"
  >
     
    corona details
  </div>,
  <ul
    className="details"
  >
    <li
      className="detailsElement"
    >
      <div>
        Country
      </div>
      <div />
    </li>
    <li
      className="detailsElement"
    >
      <div>
        Population
      </div>
      <div>
        0
      </div>
    </li>
    <li
      className="detailsElement"
    >
      <div>
        Confirmed cases
      </div>
      <div>
        0
      </div>
    </li>
    <li
      className="detailsElement"
    >
      <div>
        Critical cases
      </div>
      <div>
        0
      </div>
    </li>
    <li
      className="detailsElement"
    >
      <div>
        Confirmed deaths
      </div>
      <div>
        0
      </div>
    </li>
    <li
      className="detailsElement"
    >
      <div>
        Recovered cases
      </div>
      <div>
        0
      </div>
    </li>
  </ul>,
]
`;

exports[`Testing pages test Home page 1`] = `
<div>
  <header
    className="header"
  >
    Home
    <div>
      <input
        onChange={[Function]}
        placeholder="Search country"
        type="text"
        value={Array []}
      />
      <img
        alt="settings"
        src="settings.png"
      />
    </div>
  </header>
  <div
    className="intro"
  >
    <h1>
      Corona Virus Data
    </h1>
    <div
      className="allData"
    >
      <div>
        <div
          className="value"
        >
          1
        </div>
        <div
          className="parameter"
        >
          Number of countries
        </div>
      </div>
      <div>
        <div
          className="value"
        >
          NaN
        </div>
        <div
          className="parameter"
        >
          Total number of cases
        </div>
      </div>
      <div>
        <div
          className="value"
        >
          NaN
        </div>
        <div
          className="parameter"
        >
          Total number of deaths
        </div>
      </div>
    </div>
  </div>
  <div
    className="divide"
  >
    Stats by countries
  </div>
  <div
    className="countriesContainer"
  >
    <a
      className="link"
      href="/details/"
      onClick={[Function]}
    >
      <div
        className="container"
      >
        <div>
          <img
            alt="details"
            src="details.png"
          />
        </div>
        <div
          className="bottom"
        >
          <div
            className="name"
          />
          <div />
        </div>
      </div>
    </a>
  </div>
</div>
`;
